source/multigrid/mg_tools.cc

```diff
@@ -1547,6 +1547,42 @@ namespace MGTools
       }
 
   }
+
+
+
+  template <int dim, int spacedim>
+  unsigned int
+  max_level_for_coarse_mesh (const Triangulation<dim,spacedim> &tria)
+  {
+    // Find minimum level for an active cell in
+    // this locally owned subdomain
+    // Note: with the way active cells are traversed,
+    // the first locally owned cell we find will have
+    // the lowest level in the particular subdomain.
+    unsigned int min_level = tria.n_global_levels();
+    typename Triangulation<dim,spacedim>::active_cell_iterator cell = tria.begin_active(),
+                                                               endc = tria.end();
+    for (; cell!=endc; ++cell)
+      if (cell->is_locally_owned())
+        {
+          min_level = cell->level();
+          break;
+        }
+
+    // If necessary, communicate to find minimum
+    // level for an active cell over all subdomains
+    unsigned int global_min;
+    if (const parallel::Triangulation<dim,spacedim> *tr
+        = dynamic_cast<const parallel::Triangulation<dim,spacedim> *>
+        (&tria))
+      global_min = Utilities::MPI::min(min_level, tr->get_communicator());
+    else
+      global_min = min_level;
```

Comment:
@davydden Is this what you were thinking for communication? I was unsure, but I've seen this style elsewhere in the library.

Reply:
Yes, that’s what I meant

Reply:
Uninitialized variables smell. Here, it probably doesn't matter too much as we initialize it immediately afterwards. On the other hand, you can easily refactor this using an immediately invoked lambda into
```
const unsigned int global_min = [&]()
{
  if (const parallel::Triangulation<dim,spacedim> *tr
      = dynamic_cast<const parallel::Triangulation<dim,spacedim> *> (&tria))
    return Utilities::MPI::min(min_level, tr->get_communicator());
  return min_level;
}();
```
which even allows the variable to be `const`.

Reply:
Aw, that's unnecessarily hard to read. I think the code as above is actually quite ok given that the uninitialized variable is clearly initialized in both branches of the otherwise trivial `if-else` clause. But if you don't like the fact that the variable is uninitialized at first, just initialize it to `min_level` and omit the `else` branch.